azure-pipelines/integrationtests.yml

```diff
@@ -1,56 +1,67 @@
 pool:
-    name: SqlToolsTestAgentPool
-    demands: VisualStudio_16.0
+    name: ads-build-1es-hosted-pool
+    demands: 
+    - ImageOverride -equals ADS-Linux_Image
 
 steps:
-  - task: DownloadSecureFile@1
-    displayName: 'Download secure file'
+
+  - task: DockerInstaller@0
+    displayName: Installing Docker
     inputs:
       secureFile: 'testsettings.json'
+      dockerVersion: 17.09.0-ce
+      releaseType: stable
 
-  - task: VisualStudioTestPlatformInstaller@1
-    displayName: 'Visual Studio Test Platform Installer'
-    inputs:
-      versionSelector: latestStable
+  - script: docker pull mcr.microsoft.com/mssql/server:2019-latest
+    displayName: Pull MSSQL Docker Image  
+
+  - script: 'docker run -e "ACCEPT_EULA=Y" -e "SA_PASSWORD=$(sql-server-password)"
+    -e "MSSQL_AGENT_ENABLED=True"
+    -p 1433:1433 --name sql1 -h sql1
+    -d mcr.microsoft.com/mssql/server:2019-latest'
+    displayName: Starting Server in Docker Container
 
   - task: UseDotNet@2
-    displayName: 'Use .NET Core sdk '
+    displayName: Use .NET Core SDK
     inputs:
       useGlobalJson: true
 
+  - task: Bash@3
+    displayName: Creating Loc Directories Expected By Dotnet Build
+    inputs:
+      filePath: ./azure-pipelines/createBuildDirectories.sh
+
   - task: NuGetAuthenticate@0
 
   - task: DotNetCoreCLI@2
-    displayName: 'dotnet build'
+    displayName: Building Test Environment
     inputs:
-      projects: '**/Microsoft.SqlTools.ServiceLayer.IntegrationTests.csproj'
-      arguments: '/p:CodeCoverageBuild=true'
+      projects: '**/Microsoft.SqlTools.ServiceLayer.TestEnvConfig.csproj'
 
-  - task: CopyFiles@2
-    displayName: 'Copy testsettings file to bin'
+  - task: DownloadSecureFile@1
+    displayName: Download Test Environment Configuration
+    name: testEnvironmentPath
     inputs:
-      SourceFolder: '$(Agent.TempDirectory)'
-      Contents: '**\testsettings.json'
-      TargetFolder: '$(Build.SourcesDirectory)\test\Microsoft.SqlTools.ServiceLayer.IntegrationTests\bin\Debug\net6.0'
+      secureFile: 'SQLConnectionInstances.xml'

```

Comment:
If we're using docker do we need this?

Reply:
![image](https://user-images.githubusercontent.com/6816294/162254665-752a5882-510a-4f25-9756-78436eeb9bf7.png)
I followed this guide